docs/general/build/smart-contracts/contract-security/signature-replay.md

```diff
@@ -88,3 +88,30 @@ To prevent signature replay attacks, you can use the following methods:
    }
    ```
 
+3. **Using EIP-712 for Structured Data**:
+Implement EIP-712 to create a more secure and structured data signing experience, which helps prevent signature replays across different contexts and contracts.
+
+   ```solidity
+   // Utilizing EIP-712 to create a typed structured data signature
+   using EIP712 for bytes32;
+
+   function eip712Mint(address recipient, uint amount, bytes memory signature) public {
+       bytes32 structHash = keccak256(abi.encode(
+           keccak256("Mint(address recipient,uint256 amount,uint256 nonce)"),
+           recipient,
+           amount,
+           nonce
+       ));
+       bytes32 digest = _hashTypedDataV4(structHash);
+       require(ECDSA.recover(digest, signature) == authorizedSigner, "Invalid EIP-712 signature!");
+       _mint(recipient, amount);
+       nonce++;
+   }
+   ```
+
+You can find more detailed implementation guidelines and tools in [OpenZeppelin's EIP-712 documentation](https://docs.openzeppelin.com/contracts/5.x/api/utils#EIP712).
+
+4.**Implementing CIP-23 for Cross-Chain Safety**:
+CIP-23 is an adaptation of Ethereum's EIP-712, designed to enhance security in cross-chain operations. It introduces specific measures to prevent replay attacks by including a unique chainId in the signature data, ensuring that signatures are only valid on their intended chain. 
```

Comment:
```suggestion
CIP-23 is an adaptation of Ethereum's EIP-712 for **Conflux core space**, designed to enhance security in cross-chain operations. It introduces specific measures to prevent replay attacks, ensuring that signatures for EVM-compatible chains cannot be replayed for Conflux core space, and vice versa. 
```